pkg/drain/drain.go

```diff
@@ -153,13 +155,27 @@ func (d *Helper) GetPodsForDeletion(nodeName string) (*podDeleteList, []error) {
 				// through any additional filters
 				break
 			}
+			if status.priority < priority {
+				//a previous filter had determined that this pod should be deleted at a lower priority
+				// use that priority
+				priority = status.priority
+			}
 		}
+		status.priority = priority
 		pods = append(pods, podDelete{
 			pod:    pod,
 			status: status,
 		})
 	}
 
+	//sort the list to come up with a sensible order for deleting pods
+	//for backwards compatibility, the default priority is highest.  If a daemon-set is found
+	//and ignore-daemonsets is set to false, the priority will be the lowest.
+	//This sort will return the order of pods to delete, with the highest priority first.
+	sort.SliceStable(pods, func(i, j int) bool {
```

Comment:
`evictPods()` uses goroutines to evict all pods in parallel, so sorting the list has little effect. `DaemonSet` pods will still be evicted before the workloads that depend on them.